src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1048,6 +1048,96 @@ public void TypeDeclaration_ParameterlessConstructor_02()
                 Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "S2").WithLocation(7, 14));
         }
 
+        [Fact]
+        public void TypeDeclaration_ParameterlessConstructor_OtherConstructors()
+        {
+            var src = @"
+record struct S1
+{
+    public S1() { }
+    S1(object o) { } // ok because no record parameter list
+}
+record struct S2
+{
+    S2(object o) { }
+}
+record struct S3()
+{
+    S3(object o) { } // 1
+}
+record struct S4()
+{
+    S4(object o) : this() { }
+}
+record struct S5(object o)
+{
+    public S5() { } // 2
+}
+record struct S6(object o)
+{
+    public S6() : this(null) { }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (13,5): error CS8862: A constructor declared in a record with parameter list must have 'this' constructor initializer.
+                //     S3(object o) { } // 1
+                Diagnostic(ErrorCode.ERR_UnexpectedOrMissingConstructorInitializerInRecord, "S3").WithLocation(13, 5),
```

Comment:
Can we improve this error message? I don't understand what it's telling me.

Reply:
I suppose this is probably out of scope for this PR, but I really do find this confusing.

Reply:
As I mentioned offline, this error message correctly describes the problem. But we can certainly improve the wording separately from the parameterless struct ctor feature work if you have a suggestion.